### PR TITLE
Feat/add tsci lint

### PR DESCRIPTION
**Pull request by imrishabh18**
Lint error shown

Screenshot 2024-07-10 at 3 52 26 PM

Closes issue - #13

### CODE REVIEW

**Comment by seveibar** [comment]
So I think `tsci lint` should not require the "target project" (meaning the project the person is running `tsci dev` in) to have eslint installed (we need to bundle it). I also think we should _only run custom rules for tscircuit_ not normal linting rules. For example: If you run `tsci lint` it might say `You have a capacitor without a unit value "100", try changing to "100uF"`

**Comment by imrishabh18** [comment]
@seveibar Take a look now, I think this is a good place to start. We can either add the plugins config in this object `tscircuitPlugin`, or later publish it as a whole plugin in a separate repo. This should be done, when we have some listed rules that needs to be implemented

File: lib/cmd-fns/lint.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          description: "Ensure capacitor values include units",
+          category: "Possible Errors",
+          recommended: true,
+        },
+        fixable: "code",
+      },
+      create(context: Rule.RuleContext) {
+        return {
+          CallExpression(node: any) {
+            if (node.callee.name === 'Capacitor') {
```
Comment (seveibar) [inline lib/cmd-fns/lint.ts]:
```suggestion
            if (node.callee.name === 'capacitor') {
```

File: lib/cmd-fns/lint.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+              if (valueArg) {
+                const valueProp = valueArg.properties.find((prop: any) => prop.key.name === 'value')
+                if (valueProp && valueProp.value.type === 'Literal') {
+                  const value = valueProp.value.value
+                  if (typeof value === 'string' && !value.match(/[µuμ]F$/)) {
+                    context.report({
+                      node: valueProp,
+                      message: 'Capacitor value should include units (e.g., "100µF")',
+                      fix(fixer) {
+                        return fixer.replaceText(valueProp.value, `"${value}µF"`)
```
Comment (seveibar) [inline lib/cmd-fns/lint.ts]: technically uF changes the value by dividing by 100


```suggestion
                        return fixer.replaceText(valueProp.value, `"${value}F"`)
```

**Review by seveibar — CHANGES_REQUESTED** [review]
Ok i do like where this is going. Can you remove eslint from the top level since we use biome (biome doesnt support plugins so eslint is better for target projects)

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    plugins: ['@typescript-eslint'],
+    extends: [
+      'eslint:recommended',
+      'plugin:@typescript-eslint/recommended',
+    ],
+    ignorePatterns: ['**/node_modules/**', 'dist/**'],
+    rules: {
+      // Add any custom rules here
+    },
+  };
```
Comment (seveibar) [inline .eslintrc.js]: Can we remove this file (we use biome)

File: .eslintignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+node_modules
+dist
```
Comment (seveibar) [inline .eslintignore]: can we remove this file

File: lib/util/tscircuitLinter.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
+import * as ts from 'typescript';
```
Comment (seveibar) [inline lib/util/tscircuitLinter.ts]: the naming convention for this project is _supposed_ to be `ExportNameExact` or `kebab-case` (usually kebab-case). Also the file name should always match the export, `tscircuitLinter.ts` should be `lint-project.ts`

FWIW I generally believe these rules shouldn't be enforced if there isn't a linter running on ci... so feel free to ignore suggestion to rename

Comment (imrishabh18) [reply lib/util/tscircuitLinter.ts]: Cool! It's okay, I will rename. If the codebase is following a standard practice, then it's fine to follow it thoroughly.

**Review by seveibar — APPROVED** [review]

**Pull request merged**